common/shared.c

```diff
@@ -1021,6 +1021,9 @@ ssize_t ft_rx(struct fid_ep *ep, size_t size)
 	}
 	/* TODO: verify CQ data, if available */
 
+	/* Ignore the size arg. Otherwise posted receive wouldn't have enough space
+	 * when the benchmark tests increments the send data size. Instead we post
+	 * a size of the entire receive buffer. */
```

Comment:
@sayantansur yes, rx_size (which is the size of the recv posted here) is tx_size + ft_rx_prefix_size() and tx_size is fi->ep_attr->max_msg_size.
Also, we do have ft_sync() between message size updates in benchmarks tests. ft_sync() makes use of ft_rx() and it needs to post a buffer large enough to receive the different messages sizes that we run our tests for. I shall rephrase the comment to have this info.